L1Trigger/Phase2L1ParticleFlow/src/PUAlgoBase.cc

```diff
@@ -75,6 +95,109 @@ void PUAlgoBase::doVertexing(std::vector<Region> &rs, VertexAlgo algo, float &pv
   }
 }
 
+void PUAlgoBase::doVertexings(std::vector<Region> &rs, VertexAlgo algo, std::vector<float> &pvdz) const {
+  int lNBins = int(40. / vtxRes_);
+  if (algo == VertexAlgo::TP || algo == VertexAlgo::External)
+    lNBins *= 3;
+  std::unique_ptr<TH1F> h_dz(new TH1F("h_dz", "h_dz", lNBins, -20, 20));
+  for (const Region &r : rs) {
+    for (const PropagatedTrack &p : r.track) {
+      if (rs.size() > 1) {
+        if (!r.fiducialLocal(p.floatVtxEta(), p.floatVtxPhi()))
+          continue;  // skip duplicates
+      }
+      h_dz->Fill(p.floatDZ(), std::min(p.floatPt(), 50.f));
+    }
+  }
+  switch (algo) {
+    case VertexAlgo::External: {
+      int lBin[nVtx_];
```

Comment:
No idea, this piece of code was added by @violatingcp so he may know better, but anway we should remove this entire file in a new PR because it's obsolete